In [1]:
import pandas as pd
import numpy as np
pd.set_option('max_colwidth', 200)
pd.set_option('display.max_columns', None)

In [13]:
# Radi Bois
#trans_path = './releves/mouvement_rb_3.5-to-8.5.xls'
#trans_path = './releves/mouvement_rb_9.5-to-22.5.xls'
trans_path = './releves/mouvement_nr_10.5-to-22.5.xls'
cheques_path = './releves/nr_cheques-20.5.txt'

# Nezha
#trans_path = './releves/mouvement_nr_23.4-to-9.5.xls'
#cheques_path = './releves/nr_cheques.txt'

In [22]:
# Read data from xls file
df_trans = pd.read_csv(trans_path, 
        delimiter="\t", skiprows=8, skipfooter=2, engine='python')
# Drop date_Valeur column and Debit column
df_trans = df_trans.drop(['date_Valeur', 'Debit'], axis=1)
df_trans.Libelle = df_trans.Libelle.astype(str)

print("Bank Transactions:", df_trans.shape[0])
df_trans = df_trans.sort_values('DateOperation')
df_trans.style.set_properties(subset=["Libelle"],**{'text-align': 'left'})

Bank Transactions: 16


In [33]:
df_cheques = pd.read_csv(cheques_path, dtype=str)
df_cheques.amount = df_cheques.amount.astype(int)
print("Cheques:", df_cheques.shape[0])
df_cheques

Cheques: 21


chequeNo  amount
0   0699161       0
1   2275879       0
2   0740911       0
3   2100785       0
4   0699207       0
5   9533546       0
6   0699250       0
7   2100005       0
8   0935166       0
9   1300040       0
10  0224806       0
11  9533547       0
12  6918683       0
13  0699208       0
14  1800069       0
15  9478120       0
16  2476206       0
17  0699251       0
18  4663824       0
19  0699209       0
20  9533548       0

In [41]:
# Lookup the df_cheques.chequeNo in the df_trans.Libelle column
# and return the corresponding Libelle
#result = df_trans[df_trans['Libelle'].str.contains('|'.join(df_cheques['chequeNo'].astype(str)))] #['Libelle']
#print(result)
df_trans['chequeNo'] = df_trans['Libelle'].str.extract(r'(\d*\.\d+|\d+)', expand=False) #.fillna(0).astype(int) 

#df_cheques['chequeNo'] = df_cheques['chequeNo'].astype(int)

# Concat df_cheques with df_trans
df_merged = df_cheques.merge(df_trans, 'outer').sort_values('DateOperation')
# Select rows where amount is NaN
print("VA Cheques:")
display(df_merged[df_merged['amount'].isnull()])

# Select rows where DateOperation is NaN
print("Unpaid cheques:")
display(df_merged[df_merged['DateOperation'].isnull()]["chequeNo"])

# Select rows where DateOperation is NaN
print("Paid cheques:", df_merged[df_merged['DateOperation'].notna()].shape[0])
display(df_merged[df_merged['DateOperation'].notna()])


VA Cheques:


Empty DataFrame
Columns: [chequeNo, amount, DateOperation, Libelle, Credit]
Index: []

Unpaid cheques:


6     0699250
9     1300040
10    0224806
15    9478120
18    4663824
Name: chequeNo, dtype: object

Paid cheques: 16


chequeNo  amount DateOperation                                    Libelle  \
5   9533546       0    2024-05-10          ENC CHQ 9533546 MENNIOUI MOHAMMED   
0   0699161       0    2024-05-14        ENC CHEQUE 0699161 ATTIZITI SOFIANE   
1   2275879       0    2024-05-14  ENC CHEQUE 2275879 EL BENNAR EL HOUSSAINE   
2   0740911       0    2024-05-14          ENC CHEQUE 0740911 SLAH ABDELATIF   
3   2100785       0    2024-05-14     ENC CHEQUE 2100785 MOHAMED EL MOUIKINI   
4   0699207       0    2024-05-14        ENC CHEQUE 0699207 ATTIZITI SOFIANE   
11  9533547       0    2024-05-14          ENC CHQ 9533547 MENNIOUI MOHAMMED   
7   2100005       0    2024-05-16           ENC CHEQUE 2100005 BACHAR ISMAIL   
8   0935166       0    2024-05-16          ENC CHEQUE 0935166 BERZAKHE FOUAD   
12  6918683       0    2024-05-16         ENC CHEQUE 6918683 MOUHSINE OUAKKI   
13  0699208       0    2024-05-16        ENC CHEQUE 0699208 ATTIZITI SOFIANE   
14  1800069       0    2024-05-16          ENC CHEQUE 1800069 MOHAMMED ALAMI   
20  9533548       0    2024-05-16          ENC CHQ 9533548 MENNIOUI MOHAMMED   
16  2476206       0    2024-05-20       ENC CHEQUE 2476206 BOUARGALNE LAHCEN   
17  0699251       0    2024-05-20        ENC CHEQUE 0699251 ATTIZITI SOFIANE   
19  0699209       0    2024-05-20        ENC CHEQUE 0699209 ATTIZITI SOFIANE   

     Credit  
5   10000.0  
0   20500.0  
1   15200.0  
2   11934.0  
3   25000.0  
4   16000.0  
11  10000.0  
7    5000.0  
8   10625.0  
12  27195.0  
13  16000.0  
14  15000.0  
20  10000.0  
16   4450.0  
17  20500.0  
19  16000.0

### Mois 5 - Nezha
From Alaoui:
6     0699250 => Deposited 22.5.2024
9     1300040 => Returned then Deposited 23.5.2024
10    0224806 => Panadoor paid
15    9478120 => Panadoor paid
18    4663824 => Panadoor paid

	chequeNo	amount	DateOperation	Libelle	Credit \
8	224806	NaN	2024-05-16	ENC CHEQUE 0224806 HICHAM ANBI	23000 \
9	9478120	NaN	2024-05-16	ENC CHEQUE 9478120 ES-SALHI MOHAMED	42800 \
10	4663824	NaN	2024-05-20	ENC CHEQUE 4663824 STE AJI TRADIVE	8800 

#### 5     9433427 => Received cash 
DateOperation	Libelle	Credit	Facture No	Cheque No \
9876378 => 07/05/2024	ENC CHEQUE 9876378 MANARA POLYGRAF	20760	FA20240154	9876378 \
0891025 => 07/05/2024	ENC CHEQUE 0891025 STE PRIDELQUA MAROC SARL AU	11000	FA20240161	0891025 \
9193851 => 07/05/2024	ENC CHEQUE 9193851 NINOS SCHOOL	5000	FA20240162	9193851 


In [43]:
%pip install pyodbc

   ---------------------------------------- 0.0/69.3 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/69.3 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/69.3 kB ? eta -:--:--
   ----- ---------------------------------- 10.2/69.3 kB ? eta -:--:--
   ----------------- ---------------------- 30.7/69.3 kB 131.3 kB/s eta 0:00:01
   ----------------------------------- ---- 61.4/69.3 kB 233.8 kB/s eta 0:00:01
   ---------------------------------------- 69.3/69.3 kB 236.1 kB/s eta 0:00:00
Note: you may need to restart the kernel to use updated packages.


In [52]:

import pyodbc
import pandas as pd
# Some other example server values are
# server = 'localhost\sqlexpress' # for a named instance
# server = 'myserver,port' # to specify an alternate port
server = 'MCOEFLAE1250401' 
database = 'Panadoor' 
username = 'dbuser' 
password = 'pass123'  
conn = pyodbc.connect('DRIVER={SQL Server};SERVER='+server+';DATABASE='+database+';UID='+username+';PWD='+ password)
cursor = conn.cursor()
# select 26 rows from SQL table to insert in dataframe.
query = "Select * from dbo.F_COMPTET"
df = pd.read_sql(query, conn)
display(df.head(26))

C:\Users\erradi\AppData\Local\Temp\ipykernel_17292\1110947426.py:14: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df = pd.read_sql(query, conn)


CT_Num     cbCT_Num                       CT_Intitule  CT_Type  \
0   44110001  b'44110001'    Emirates Panel Manufacture LLC        1   
1   34210001  b'34210001'                   Ste Netale Allu        0   
2   34210002  b'34210002'              Ste Tiziti Aluminium        0   
3   34210003  b'34210003'            Ste Verre Oulad Chaker        0   
4   34210004  b'34210004'                    Hamid El Hasny        0   
5   34210005  b'34210005'                   Mouhsine Ouakki        0   
6   34210006  b'34210006'               Abdechafiq Massoudi        0   
7   34210007  b'34210007'                   Abdelaziz Omari        0   
8   34210008  b'34210008'                 Mohamed El Ouardy        0   
9   34210009  b'34210009'                  El Arabi El Abid        0   
10  34210010  b'34210010'                    Ste Delta Line        0   
11  34210011  b'34210011'                     Ste Keniglass        0   
12  34210012  b'34210012'                    Ste Promo Saad        0   
13  34210013  b'34210013'                  Ste Frolum Group        0   
14  34210014  b'34210014'               Ste Verre Acherafae        0   
15  34210015  b'34210015'                   Ste Guedira.Tex        0   
16  34210016  b'34210016'                      Ste Lina Sud        0   
17  34210017  b'34210017'  Ste Industrie Marocaine Du Verre        0   
18  34210018  b'34210018'               Ste Nord 2P Travaux        0   
19  34210019  b'34210019'                      Ste Sogercos        0   
20  44110002  b'44110002'              Carburo del Cinca SA        1   
21  34210020  b'34210020'   Ste Construction Euro Catalonia        0   
22  34210021  b'34210021'                        Jamal Frit        0   
23  34210022  b'34210022'       Ste Belmekki Hassan et Fils        0   
24  34210023  b'34210023'                      Khalid Alami        0   
25  34210024  b'34210024'                            Nejjar        0   

   CG_NumPrinc cbCG_NumPrinc CT_Qualite      CT_Classement  \
0     44110000   b'44110000'                           EPM   
1     34210000   b'34210000'                        Netale   
2     34210000   b'34210000'             Ste Tiziti Alumin   
3     34210000   b'34210000'                  Oulad Chaker   
4     34210000   b'34210000'                         Hasny   
5     34210000   b'34210000'               Mouhsine Ouakki   
6     34210000   b'34210000'                      Massoudi   
7     34210000   b'34210000'                         Omari   
8     34210000   b'34210000'                        Ouardy   
9     34210000   b'34210000'              El Arabi El Abid   
10    34210000   b'34210000'                         Delta   
11    34210000   b'34210000'                 Ste Keniglass   
12    34210000   b'34210000'                         Promo   
13    34210000   b'34210000'                        Frolum   
14    34210000   b'34210000'             St Verre Acherafa   
15    34210000   b'34210000'               Ste Guedira Tex   
16    34210000   b'34210000'                  Ste Lina Sud   
17    34210000   b'34210000'             Industrie DuVerre   
18    34210000   b'34210000'               Nord 2P Travaux   
19    34210000   b'34210000'                      Sogercos   
20    44110000   b'44110000'                       Carburo   
21    34210000   b'34210000'              ConstructionEuro   
22    34210000   b'34210000'                    Jamal Frit   
23    34210000   b'34210000'                      Belmekki   
24    34210000   b'34210000'                  Khalid Alami   
25    34210000   b'34210000'                        Nejjar   

         cbCT_Classement CT_Contact CT_Adresse CT_Complement CT_CodePostal  \
0                 b'EPM'                                                     
1              b'Netale'                                                     
2   b'Ste Tiziti Alumin'                                                     
3        b'Oulad Chaker'                                                  